python/ray/workflow/step_executor.py

```diff
@@ -209,16 +227,22 @@ async def _write_step_inputs(wf_storage: workflow_storage.WorkflowStorage,
 
 
 def commit_step(store: workflow_storage.WorkflowStorage, step_id: "StepID",
-                ret: Union["Workflow", Any], *,
-                exception: Optional[Exception]):
+                ret: Union["Workflow", Any], *, exception: Optional[Exception],
+                checkpoint: "CheckpointModeType"):
     """Checkpoint the step output.
     Args:
         store: The storage the current workflow is using.
         step_id: The ID of the step.
         ret: The returned object of the workflow step.
         exception: The exception caught by the step.
+        checkpoint: The checkpoint mode.
     """
     from ray.workflow.common import Workflow
+
+    if checkpoint == CheckpointMode.SKIP.value:
+        return
+    elif checkpoint != CheckpointMode.SYNC.value:
+        raise ValueError(f"Unknown checkpoint mode: {checkpoint}.")
```

Comment:
Let's don't do checkpoint logic here inside commit_step, it just seems wired, because you are checkpointing things, but actually, it does nothing. So the function is not doing what it's supposed to do.

Let's move the logic to the upper layer.